quiz/quiz.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bufio"
+	"encoding/csv"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"time"
+	"unicode"
+)
+
+type problems struct {
+	q string
+	a string
+}
+
+func main() {
+	const defaultName = "problems.csv"
+	const defaultTimeLimit = 30
+	fileName := flag.String("csv", defaultName, "csv file in the format: `question,answer`")
+	timeLimit := flag.Int("limit", defaultTimeLimit, "Integer time limit ")
+	flag.Parse()
+
+	if isValidFileName(*fileName) {
+		arr, err := readFile(*fileName)
+		if err != nil {
+			log.Fatal(err)
+			return
+		}
+		problems := parseQuiz(arr)
+		quiz(problems, *timeLimit)
+	} else {
+		log.Fatal("Invalid file name")
+		return
+	}
+}
+
+func quiz(problems []problems, timeLimit int) {
+	correct := 0
+	incorrect := 0
+	timer := time.NewTimer(time.Duration(timeLimit) * time.Second)
+
+	for idx, prob := range problems {
+		fmt.Printf("Question number %d is %s? \n", idx+1, prob.q)
+		select {
+		case <-timer.C:
+			fmt.Println("Ops! You ran out of time")
+			fmt.Printf("out of %d questions, you got %d questions correct \n", len(problems), correct)
+			return
+		default:
+			var response string
+			fmt.Scanln(&response)
+			if response == prob.a {
+				fmt.Println("Genius! You got the answer right")
+				correct++
+			} else {
+				fmt.Println("You failed that question but that doesn't make you a failure :)")
+				incorrect++
+			}
+		}
+	}
+	fmt.Printf("out of %d questions, you got %d questions correct \n", len(problems), correct)
+	return
+}
+
+func parseQuiz(arr [][]string) []problems {
+	ret := make([]problems, len(arr))
+	for i, row := range arr {
+		ret[i] = problems{
+			q: row[0],
+			a: row[1],
+		}
+	}
+	return ret
+}
+
+func readFile(fileName string) ([][]string, error) {
+	problems, err := os.Open(fileName)
+	if err != nil {
+		fmt.Printf("Failed to open file with filename %s \n", fileName)
+		os.Exit(1)
+	}
+	header, err := bufio.NewReader(problems).ReadSlice('\n')
+	if err != nil {
+		return nil, err
+	}
+	_, err = problems.Seek(int64(len(header)), io.SeekStart)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why do you need to do this? 👀

Reply:
This was just an attempt to understand how to start from a different line. In my head, the CSV file will have a header and I don't want that header to be part of the problems sent to the user. I should probably do a check for it though